src/python/aqueduct_executor/operators/param_executor/main.py

```diff
@@ -16,11 +16,14 @@ def run(spec: spec.ParamSpec) -> None:
     try:
         utils.write_artifact(
             storage,
+            enums.OutputArtifactType.JSON,
             spec.output_content_path,
             spec.output_metadata_path,
             spec.val,
-            {},
-            enums.OutputArtifactType.JSON,
+            system_metadata={},
+            user_defined_metadata={
+                constants.PARAM_NAME_KEY: spec.param_name,
```

Comment:
I don't fully understand the motivation passing around parameter name as an additional field in artifact metadata, as I thought the param name should be consistent with the param artifact name, and that's something we would want to pass around during execution. Happy to chat offline

Reply:
Discussed offline. We added the parameter names to the ExtractSpec itself instead of metadata.